worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/FlushTask.scala

```diff
@@ -101,15 +102,32 @@ private[worker] class HdfsFlushTask(
     val path: Path,
     notifier: FlushNotifier,
     keepBuffer: Boolean,
-    source: AbstractSource) extends DfsFlushTask(buffer, notifier, keepBuffer, source) {
+    source: AbstractSource,
+    finalFlush: Boolean) extends DfsFlushTask(buffer, notifier, keepBuffer, source) {
   override def flush(copyBytes: Array[Byte]): Unit = {
     val readableBytes = buffer.readableBytes()
     val hadoopFs = StorageManager.hadoopFs.get(Type.HDFS)
-    val hdfsStream = hadoopFs.append(path, 256 * 1024)
-    flush(hdfsStream) {
-      hdfsStream.write(convertBufferToBytes(buffer, copyBytes, readableBytes))
-      source.incCounter(WorkerSource.HDFS_FLUSH_COUNT)
-      source.incCounter(WorkerSource.HDFS_FLUSH_SIZE, readableBytes)
+    if (StorageManager.streamsManager != null) {
```

Comment:
Two branches of this if block have duplicate code; try to simplify this if block.
You can move hdfs stream forward and reuse it.